### PR TITLE
Implement window adaptation

### PR DESCRIPTION
**Pull request by rlouf**
I am opening a new PR based on my fork of the repo, with the latest changes.

### CODE REVIEW

File: aehmc/proposals.py [hunk 1]
```diff
@@ -87,15 +87,12 @@ def progressive_uniform_sampling(
     state, energy, weight, _ = proposal
     new_state, new_energy, new_weight, _ = new_proposal
 
-<<<<<<< HEAD
```
Comment (twiecki) [inline aehmc/proposals.py]: 🧐

Comment (rlouf) [reply aehmc/proposals.py]: Yeah I wish the code got to that point in the warmup.

File: tests/test_hmc.py [hunk 2]
```diff
@@ -39,38 +32,47 @@ def kernel_factory(inverse_mass_matrix: TensorVariable):
...
     )
 
     # mode = aesara.compile.Mode()
     # mode = mode.excluding("merge")
     trajectory_generator = aesara.function(
-        (y_vv,), trajectory[0], updates=updates  # , mode=mode
+        (), trajectory[0], updates=updates  # , mode=mode
     )
 
     print("compiled!")
```
Comment (brandonwillard) [inline tests/test_hmc.py:70]: Did this sneak in from another PR?

Comment (rlouf) [reply tests/test_hmc.py]: Yep, that's what happened.

**Review by brandonwillard — COMMENTED** [review]
It looks like `test_nuts_warmup` no longer takes an unreasonable amount of time to compile.  Has the underlying issue been fixed from here?  It definitely looked like the old code was producing unreasonably large graphs, and I'm assuming that is no longer the case, no?

**Comment by rlouf** [comment]
Ok let's clean and merge this.

**Review by brandonwillard — COMMENTED** [review]
These commits are looking pretty good now; thanks.

File: aehmc/window_adaptation.py [hunk 3]
```diff
@@ -1,4 +1,211 @@
...
+        kernel_factory, is_mass_matrix_full, initial_step_size, target_acceptance_rate
+    )
+
+    def one_step(
+        stage,  # schedule
+        is_middle_window_end,
+        q,  # chain state
+        potential_energy,
+        potential_energy_grad,
+        step,  # Dual Averagin adaptation state
```
Comment (brandonwillard) [inline aehmc/window_adaptation.py:33]:
```suggestion
        step,  # Dual average adaptation state
```

**Review by brandonwillard — COMMENTED** [review]
Aside from some potentially missing coverage, this looks good.

File: aehmc/step_size.py [hunk 4]
```diff
@@ -73,91 +70,23 @@ def dual_averaging_adaptation(
             adaptively setting path lengths in Hamiltonian Monte Carlo." Journal
             of Machine Learning Research 15.1 (2014): 1593-1623.
     """
+    da_init, da_update = algorithms.dual_averaging(gamma, t0, kappa)
 
-    mu = at.log(10) + initial_log_step_size
-    da_init, da_update = algorithms.dual_averaging(mu, gamma, t0, kappa)
+    def init(initial_step_size: TensorVariable):
+        mu = at.log(10) + at.log(initial_step_size)
+        return da_init(mu)
```
[the remark this answers is not in the document]
Comment (rlouf) [reply aehmc/step_size.py:77]: It's weird because the function is used in `test_step_size.py`.

**Review by brandonwillard — APPROVED** [review]

**Pull request merged**